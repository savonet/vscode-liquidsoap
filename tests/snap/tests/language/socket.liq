def f() =
  test.equals(socket.internet_address.ipv6.any.is_ipv6, true)
  test.equals(socket.internet_address.ipv6.loopback.to_string(), "::1")

  let [h] = null.get(host.of_name("localhost")).addresses
  test.equals(socket.address.internet_address(h, 80).port, 80)

  fname = file.temp("unix", "socket")
  file.remove(fname)

  thread.run(fun () -> begin
    s = socket.unix.listen(fname)
    test.equals(s.type, "unix")
    s.write.wait(fun () -> begin
      s.write("done!")
      s.close()
    end)
  end)

  thread.run(delay=0.1, fun () -> begin
    s = socket.unix.client(fname)
    s.read.wait(fun () ->
     begin
      if s.read() == "done!" then
        s.close()
        test.pass()
      else
        test.fail()
      end
     end)
  end)
end

test.check(f)
