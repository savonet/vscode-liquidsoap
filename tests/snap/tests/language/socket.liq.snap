>def f() =
#^^^ source.liquidsoap keyword.control.liquidsoap
#   ^^^^^^^ source.liquidsoap
>  test.equals(socket.internet_address.ipv6.any.is_ipv6, true)
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                                         ^ source.liquidsoap constant.numeric.integer
#                                          ^^^^^^^^^^^ source.liquidsoap
#                                                     ^ source.liquidsoap constant.numeric.integer
#                                                      ^^^^^^^^ source.liquidsoap
>  test.equals(socket.internet_address.ipv6.loopback.to_string(), "::1")
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                                         ^ source.liquidsoap constant.numeric.integer
#                                          ^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                                                                 ^ source.liquidsoap string.quoted.double.liquidsoap
#                                                                  ^^^ source.liquidsoap string.quoted.double.liquidsoap
#                                                                     ^ source.liquidsoap string.quoted.double.liquidsoap
#                                                                      ^^ source.liquidsoap
>
>  let [h] = null.get(host.of_name("localhost")).addresses
#^^ source.liquidsoap
#  ^^^ source.liquidsoap keyword.control.liquidsoap
#     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                                  ^ source.liquidsoap string.quoted.double.liquidsoap
#                                   ^^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                                            ^ source.liquidsoap string.quoted.double.liquidsoap
#                                             ^^^^^^^^^^^^^ source.liquidsoap
>  test.equals(socket.address.internet_address(h, 80).port, 80)
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                                                 ^^ source.liquidsoap constant.numeric.integer
#                                                   ^^^^^^^^ source.liquidsoap
#                                                           ^^ source.liquidsoap constant.numeric.integer
#                                                             ^^ source.liquidsoap
>
>  fname = file.temp("unix", "socket")
#^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                    ^ source.liquidsoap string.quoted.double.liquidsoap
#                     ^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                         ^ source.liquidsoap string.quoted.double.liquidsoap
#                          ^^ source.liquidsoap
#                            ^ source.liquidsoap string.quoted.double.liquidsoap
#                             ^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                                   ^ source.liquidsoap string.quoted.double.liquidsoap
#                                    ^^ source.liquidsoap
>  file.remove(fname)
#^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>
>  thread.run(fun () -> begin
#^^^^^^^^^^^^^ source.liquidsoap
#             ^^^ source.liquidsoap keyword.control.liquidsoap
#                ^^^^^^^ source.liquidsoap
#                       ^^^^^ source.liquidsoap keyword.control.liquidsoap
>    s = socket.unix.listen(fname)
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>    test.equals(s.type, "unix")
#^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                        ^ source.liquidsoap string.quoted.double.liquidsoap
#                         ^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                             ^ source.liquidsoap string.quoted.double.liquidsoap
#                              ^^ source.liquidsoap
>    s.write.wait(fun () -> begin
#^^^^^^^^^^^^^^^^^ source.liquidsoap
#                 ^^^ source.liquidsoap keyword.control.liquidsoap
#                    ^^^^^^^ source.liquidsoap
#                           ^^^^^ source.liquidsoap keyword.control.liquidsoap
>      s.write("done!")
#^^^^^^^^^^^^^^ source.liquidsoap
#              ^ source.liquidsoap string.quoted.double.liquidsoap
#               ^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                    ^ source.liquidsoap string.quoted.double.liquidsoap
#                     ^^ source.liquidsoap
>      s.close()
#^^^^^^^^^^^^^^^^ source.liquidsoap
>    end)
#^^^^ source.liquidsoap
#    ^^^ source.liquidsoap keyword.control.liquidsoap
#       ^^ source.liquidsoap
>  end)
#^^ source.liquidsoap
#  ^^^ source.liquidsoap keyword.control.liquidsoap
#     ^^ source.liquidsoap
>
>  thread.run(delay=0.1, fun () -> begin
#^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                   ^^^ source.liquidsoap constant.numeric.float
#                      ^^ source.liquidsoap
#                        ^^^ source.liquidsoap keyword.control.liquidsoap
#                           ^^^^^^^ source.liquidsoap
#                                  ^^^^^ source.liquidsoap keyword.control.liquidsoap
>    s = socket.unix.client(fname)
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>    s.read.wait(fun () ->
#^^^^^^^^^^^^^^^^ source.liquidsoap
#                ^^^ source.liquidsoap keyword.control.liquidsoap
#                   ^^^^^^^ source.liquidsoap
>     begin
#^^^^^ source.liquidsoap
#     ^^^^^ source.liquidsoap keyword.control.liquidsoap
>      if s.read() == "done!" then
#^^^^^^ source.liquidsoap
#      ^^ source.liquidsoap keyword.control.liquidsoap
#        ^^^^^^^^^^^^^ source.liquidsoap
#                     ^ source.liquidsoap string.quoted.double.liquidsoap
#                      ^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                           ^ source.liquidsoap string.quoted.double.liquidsoap
#                            ^ source.liquidsoap
#                             ^^^^ source.liquidsoap keyword.control.liquidsoap
>        s.close()
#^^^^^^^^^^^^^^^^^^ source.liquidsoap
>        test.pass()
#^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>      else
#^^^^^^ source.liquidsoap
#      ^^^^ source.liquidsoap keyword.control.liquidsoap
>        test.fail()
#^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>      end
#^^^^^^ source.liquidsoap
#      ^^^ source.liquidsoap keyword.control.liquidsoap
>     end)
#^^^^^ source.liquidsoap
#     ^^^ source.liquidsoap keyword.control.liquidsoap
#        ^^ source.liquidsoap
>  end)
#^^ source.liquidsoap
#  ^^^ source.liquidsoap keyword.control.liquidsoap
#     ^^ source.liquidsoap
>end
#^^^ source.liquidsoap keyword.control.liquidsoap
>
>test.check(f)
#^^^^^^^^^^^^^^ source.liquidsoap
>